### PR TITLE
Add run_if condition in jobs

### PR DESCRIPTION
**Pull request by KoningJasper**
Add support for private preview functionality `run_if`.

### CODE REVIEW

File: jobs/resource_job.go [hunk 1]
```diff
@@ -153,6 +153,10 @@ type JobTaskSettings struct {
 	Description string           `json:"description,omitempty"`
 	DependsOn   []TaskDependency `json:"depends_on,omitempty"`
 
+	// BEGIN Jobs + RunIf preview
```
Comment (alexott) [inline jobs/resource_job.go:156]: will it be used for some automation later?

**Review by alexott — CHANGES_REQUESTED** [review]
please run `make fmt`

File: jobs/resource_job.go (hunk 1 above)
Comment (KoningJasper) [reply jobs/resource_job.go:156]: I saw it on other parts, not sure what it's for.

Comment (alexott) [reply jobs/resource_job.go:156]: it's not strictly required...

**Pull request merged**